locking/locks.go

```diff
@@ -225,33 +249,101 @@ func (c *Client) lockIdFromPath(path string) (string, error) {
 	}
 }
 
+// We want to use a single cache file for integrity, but to make it easy to
+// list all locks, prefix the id->path map in a way we can identify (something
+// that won't be in a path)
+const idKeyPrefix = "*id*://"
+
+func (c *Client) encodeIdKey(id string) string {
```

Comment:
Not sure if `(c *Client)` is the right receiver for this method. What do you think about wrapping `tools/kv` in this package with functions that understand `*Lock` and etc, and putting this method there?

```go
package locking

const (
       idPrefix string = "*id*://"
)

type LockCache struct {
        kv *kv.Store
}

func (c *LockCache) CacheLock(l Lock) error {
        // ...
}

func (c *LockCache) encodeId(id string) string {
       if !c.isId(id) {
               return idPrefix+id
        }

       return id
}

func (c *LockCache) isId(key string) bool {
       return strings.HasPrefix(key, idPrefix)
}
```